chains/evm/deployment/v1_7_0/sequences/deploy_chain_contracts.go

```diff
@@ -355,20 +365,56 @@ var DeployChainContracts = cldf_ops.NewSequence(
 	},
 )
 
-func maybeDeployContract[ARGS any](
-	b cldf_ops.Bundle,
-	op *cldf_ops.Operation[contract.DeployInput[ARGS], datastore.AddressRef, evm.Chain],
-	chain evm.Chain,
-	input contract.DeployInput[ARGS],
-	existingAddresses []datastore.AddressRef) (datastore.AddressRef, error) {
-	for _, ref := range existingAddresses {
-		if ref.Type == datastore.ContractType(input.TypeAndVersion.Type) {
-			return ref, nil
+// getMockReceiverVerifiers finds the required and optional verifier addresses given the mock receiver
+// params, the addresses of the newly deployed contracts, and the addresses of the existing contracts.
+func GetMockReceiverVerifiers(
```

Comment:
Would like to keep this function private if possible, only want sequences exported here. If it has to be public for test purposes maybe we can move it to a different folder?